src/deferred.js

```diff
@@ -345,14 +346,11 @@ jQuery.extend( {
 						updateFunc( i, progressContexts, progressValues )
 					);
 				} else {
-					--remaining;
+					updateFunc( i, resolveContexts, resolveValues )( resolveValues[ i ] );
 				}
 			}
-		}
-
-		// If we're not waiting on anything, resolve the master
-		if ( !remaining ) {
-			master.resolveWith( resolveContexts, resolveValues );
+		} else {
+			master.resolveWith( resolveContexts );
```

Comment:
`resolveContexts` is guaranteed to be `undefined` here... was it smaller to keep it in?